node/rpc/src/mpool_api.rs

```diff
@@ -0,0 +1,176 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::RpcState;
+
+use address::Address;
+use blocks::{BlockHeader, Tipset, TipsetKeys};
+use blockstore::BlockStore;
+use cid::json::CidJson;
+use encoding::Cbor;
+use jsonrpc_v2::{Data, Error as JsonRpcError, Params};
+use message::Message;
+use message::{
+    signed_message::{self, json::SignedMessageJson},
+    unsigned_message::json::UnsignedMessageJson,
+    SignedMessage,
+};
+use message_pool::*;
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use std::str::FromStr;
+use wallet::KeyStore;
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct Pending {
+    #[serde(rename = "Messages", with = "signed_message::json::vec")]
+    pub msgs: Vec<SignedMessage>,
+}
+
+/// Estimate the gas price for an Address
+pub(crate) async fn estimate_gas_price<DB, KS, MP>(
+    data: Data<RpcState<DB, KS, MP>>,
+    Params(params): Params<(u64, String, u64, TipsetKeys)>,
+) -> Result<String, JsonRpcError>
+where
+    DB: BlockStore + Send + Sync + 'static,
+    KS: KeyStore + Send + Sync + 'static,
+    MP: Provider + Send + Sync + 'static,
+{
+    let (nblocks, sender_str, gas_limit, tsk) = params;
+    let sender = Address::from_str(&sender_str)?;
+    let price = data
+        .mpool
+        .estimate_gas_price(nblocks, sender, gas_limit, tsk)?;
+    Ok(price.to_string())
+}
+
+/// get the sequence of given address in mpool
+pub(crate) async fn get_sequence<DB, KS, MP>(
+    data: Data<RpcState<DB, KS, MP>>,
+    Params(params): Params<(String,)>,
+) -> Result<u64, JsonRpcError>
+where
+    DB: BlockStore + Send + Sync + 'static,
+    KS: KeyStore + Send + Sync + 'static,
+    MP: Provider + Send + Sync + 'static,
+{
+    let (addr_str,) = params;
+    let address = Address::from_str(&addr_str)?;
+    let sequence = data.mpool.get_sequence(&address).await?;
+    Ok(sequence)
+}
+
+/// Return Vec of pending messages in mpool
+pub(crate) async fn pending<DB, KS, MP>(
+    data: Data<RpcState<DB, KS, MP>>,
+    Params(params): Params<(TipsetKeys,)>,
+) -> Result<Pending, JsonRpcError>
```

Comment:
Don't really understand why you create this new struct Pending... Seems like this method should return a list of signed messages

Reply:
yeah, that should be removed, will add change in next push